### PR TITLE
Limit scope of flux and flux operator in helm chart.

### PR DESCRIPTION
**Pull request by arturo-c**
Allows you to set `clusterScope` to `false` in helm chart so that rbac resources and arguments to both flux and helm operator are scoped only to the namespace where flux is deployed.

Partly addresses https://github.com/weaveworks/flux/issues/1886 and https://github.com/weaveworks/flux/issues/1085

### CODE REVIEW

**Comment by arturo-c** [comment]
Similar to implementation in https://github.com/zeeZ/locked-down-flux

File: chart/flux/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 appVersion: "1.12.0"
-version: 0.9.0
+version: 0.10.0
```
Comment (stefanprodan) [inline chart/flux/Chart.yaml]: Please undo this change. The chart version will be bumped in a separate PR e.g. https://github.com/weaveworks/flux/pull/1924/files

File: chart/flux/templates/deployment.yaml [hunk 2]
```diff
@@ -131,6 +131,9 @@ spec:
           {{- if .Values.memcached.createClusterIP }}
           - --memcached-service=
           {{- end }}
+          {{- if not .Values.clusterScope }}
+          - --k8s-namespace-whitelist={{ .Release.Namespace }}
```
Comment (stefanprodan) [inline chart/flux/templates/deployment.yaml]: The whitelist flag has been deprecated, please use `--k8s-allow-namespace` https://github.com/weaveworks/flux/blob/master/cmd/fluxd/main.go#L166

File: chart/flux/README.md [hunk 3]
```diff
@@ -197,6 +197,7 @@ The following tables lists the configurable parameters of the Weave Flux chart a
 | `token`                                           | `None`                                               | Weave Cloud service token
 | `extraEnvs`                                       | `[]`                                                 | Extra environment variables for the Flux pod(s)
 | `rbac.create`                                     | `true`                                               | If `true`, create and use RBAC resources
+| `clusterScope`                                    | `true`                                               | If `false`, will only create rbac resources for the namespace where flux is deployed to and will set k8s-whitelist-namespace arg in flux and allowNamespace arg in flux-operator to that namespace
```
Comment (2opremio) [inline chart/flux/README.md]: `--k8s-whitelist-namespace` is deprecated now, it's `--k8s-allow-namespace`

Comment (arturo-c) [reply chart/flux/README.md]: cool, fixed this now

File: chart/flux/Chart.yaml (hunk 1 above)
Comment (arturo-c) [reply chart/flux/Chart.yaml]: done

File: chart/flux/templates/deployment.yaml (hunk 2 above)
Comment (arturo-c) [reply chart/flux/templates/deployment.yaml]: changed

**Comment by arturo-c** [comment]
thanks @stefanprodan and @2opremio for the really quick review!, updated branch with the changes requested

**Comment by arturo-c** [comment]
also maybe @zeeZ can take a look, since thats where I found the rbac example for flux -> https://github.com/weaveworks/flux/issues/1830

**Comment by arturo-c** [comment]
@stefanprodan have not tested by having resources out of the namespace for each operator, will try and do that today and report back

File: chart/flux/templates/rbac.yaml [hunk 4]
```diff
@@ -1,8 +1,18 @@
+{{- $kind := "Role" -}}
+{{- $bind := "RoleBinding" -}}
+{{- if .Values.clusterScope -}}
+{{ $kind = "ClusterRole" }}
```
Comment (dranner-bgt) [inline chart/flux/templates/rbac.yaml]: I had to change the assignments in lines 4 and 5 to `:=` from `=` 
```suggestion
{{ $kind := "ClusterRole" }}
```

Maybe my helm client is too old?
```
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
Server: &version.Version{SemVer:"v2.13.0", GitCommit:"79d07943b03aea2b76c12644b4b54733bc5958d6", GitTreeState:"clean"}
```

Comment (dranner-bgt) [reply chart/flux/templates/rbac.yaml]: I also had to remove the `apiGroups` from the `Role` (apparently they only work for `ClusterRoles`):

```
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: {{ $kind }}
metadata:
  name: {{ template "flux.fullname" . }}
  {{- if not .Values.clusterScope }}
  namespace: {{ .Release.Namespace }}
  {{- end }}
  labels:
    app: {{ template "flux.name" . }}
    chart: {{ template "flux.chart" . }}
    release: {{ .Release.Name }}
    heritage: {{ .Release.Service }}
rules:
  - apiGroups: ["*"]
    resources: ["*"]
    verbs: ["*"]
```

Comment (dranner-bgt) [reply chart/flux/templates/rbac.yaml]: Thank you for this effort. I'm using this PR as a base to deploy flux into a cluster with access to a limited list of namespaces. I think it might be best to create a new PR that contains my additional RBAC resources once this PR is merged.

**Comment by arturo-c** [comment]
@2opremio sorry, haven't had time to work on this, not sure if best path forward is to close this pr?

**Comment by stefanprodan** [comment]
Closing this in favour of #2206
